### PR TITLE
add checkpoint to include Kubernetes cluster information

### LINKED ISSUE #3324: Extend checkpoint to include information about Kubernetes clusters
Currently, we do not have any coarse grained information about Kubernetes clusters which are powered by Weave Net. The information would be used for prioritizing issues.

1. Kubernetes version.
2. Nodes count.
3. Whether NetworkPolicies are used (and what type - Ingress / Egress).

### PR DESCRIPTION
**Pull request by murali-reddy**
Fixes #3324

### CODE REVIEW

File: prog/weaver/checkpoint.go [hunk 1]
```diff
@@ -48,10 +55,11 @@ func checkForUpdates(dockerVersion string, router *weave.NetworkRouter) {
 	}
 	kernelVersion := string(releaseBytes[:i])
 
-	flags := map[string]string{
-		"docker-version": dockerVersion,
-		"kernel-version": kernelVersion,
-	}
+	flags := make(map[string]string)
```
Comment (bboreham) [inline prog/weaver/checkpoint.go]: I don't understand the need for this change.

File: prog/weaver/checkpoint.go [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
 
 const (
-	updateCheckPeriod = 6 * time.Hour
+	updateCheckPeriod  = 6 * time.Hour
+	configMapName      = "weave-net"
+	configMapNamespace = "kube-system"
 )
 
-func checkForUpdates(dockerVersion string, router *weave.NetworkRouter) {
+func checkForUpdates(dockerVersion string, router *weave.NetworkRouter, peers []string) {
```
Comment (bboreham) [inline prog/weaver/checkpoint.go]: I think I would just pass in `len(peers)`, on the principle of minimum information.

Comment (murali-reddy) [reply prog/weaver/checkpoint.go]: Modified.

File: prog/weaver/checkpoint.go (hunk 1 above)
Comment (murali-reddy) [reply prog/weaver/checkpoint.go]: Reverted this change. Not required.

File: prog/weaver/main.go [hunk 3]
```diff
@@ -687,7 +687,7 @@ func determineQuorum(initPeerCountFlag int, router *weave.NetworkRouter) uint {
 	// that resolve to the same peer, in which case the quorum
 	// might be larger than it needs to be.  But the user can
 	// specify it explicitly if that becomes a problem.
-	clusterSize := uint(len(peers) + 1)
+	clusterSize := uint(len(peers))
```
Comment (bboreham) [inline prog/weaver/main.go]: Why did we lose the `+1` here?  The comment suggests it is the "conservative assumption"

Comment (murali-reddy) [reply prog/weaver/main.go]: :man_facepalming: My bad. Unintended.

Actually wanted to remove in the call checkForUpdates. For this call len(peers) should be sufficient right?
``` go
checkForUpdates(dockerVersion, router, uint(len(peers)+1)
```

Comment (bboreham) [reply prog/weaver/main.go]: Yeah, I would leave off the `+1` there.

**Pull request merged**